app/models/topic.rb

```diff
@@ -1632,6 +1632,36 @@ def access_topic_via_group
       .first
   end
 
+  def incoming_email_addresses(group: nil, received_before: Time.zone.now)
+    email_addresses = group.present? ? Set[group.email_username] : Set.new
+
+    # TODO(martin) Look at improving this N1, it will just get slower the
+    # more replies/incoming emails there are for the topic.
+    self.incoming_email.where("created_at <= ?", received_before).each do |incoming_email|
+      to_addresses = incoming_email.to_addresses_split
+      cc_addresses = incoming_email.cc_addresses_split
+      combined_addresses = [to_addresses, cc_addresses].flatten
+
+      # We only care about the emails addressed to the group or CC'd to the
+      # group if the group is present.
+      if group.present? && combined_addresses.any?
```

Comment:
Can you combine these conditions with the ones below? I find it a bit weird that you check here for `any?` and then for `none?`

Maybe something like this? This way you also avoid `combined_addresses`, which seems to be an unnecessary computation.

```suggestion
      next if group.present? && (
        to_addresses.any? { |address| address =~ group.email_username_regex } ||
        cc_addresses.any? { |address| address =~ group.email_username_regex }
      )
```

Reply:
What you suggested here is similar to the original code but I found the original code unclear. Also the check suggested here is wrong, we have to go next if none of the emails match the group username. I had to add .any? because I made it so the addresses would always be arrays and could not be nil. If you think the extra computation is unnecessary I will change this section back to the original, I can live with it :)

Reply:
You can leave it as it is. I do not have a strong opinion about it. :)